### ISSUE TITLE
Plugin workflow - unable to find build

### ISSUE THREAD

**Issue by xrep** [markstory/asset_compress#297]
Hi, 

at first thanks you guys for all your work. 

I am trying to build some dev. workflow, where I have plugin called "Admin", and in `app/plugins/Admin/config/asset_compress.ini `

```
[js]
timestamp = true
paths[] = node_modules/*
cachePath = WEBROOT/cached/

[base-libs.js]
files[] = jquery/dist/jquery.min.js
```

i can build this through console, after that file `app\webroot\cached\Admin.base-libs.<timestamp>.js` is created. 

The problem happens when I am trying in 

app\plugins\Admin\src\Template\Layout\default.ctp use this line:

`<?= $this->AssetCompress->script('base-libs.js'); ?>`

where I get "Cannot create a script tag for a 'base-libs.js'. That build is not defined."

or this

`<?= $this->AssetCompress->script('Admin.base-libs.js'); ?>`

where I get "Could not locate jquery/dist/jquery.min.js for Admin.base-libs.js in any configured path."

**Comment by markstory** [comment]
`Admin.base-libs.js` is the right name for the build, based on it being in a plugin. What is the stack trace for that error?

**Comment by xrep** [comment]
Hello Mark, 

heres the stacktrace:

```
⟩ MiniAsset\Factory->target
ROOT/vendor/markstory/mini-asset/src/AssetCollection.php, line 100
⟩ MiniAsset\AssetCollection->get
ROOT/vendor/markstory/asset_compress/src/View/Helper/AssetCompressHelper.php, line 287
⟩ AssetCompress\View\Helper\AssetCompressHelper->url
ROOT/vendor/markstory/asset_compress/src/View/Helper/AssetCompressHelper.php, line 234
⟩ AssetCompress\View\Helper\AssetCompressHelper->script
ROOT/plugins/Admin/src/Template/Layout/default.ctp, line 600
⟩ include
CORE/src/View/View.php, line 1016
⟩ Cake\View\View->_evaluate
CORE/src/View/View.php, line 975
⟩ Cake\View\View->_render
CORE/src/View/View.php, line 646
⟩ Cake\View\View->renderLayout
CORE/src/View/View.php, line 604
⟩ Cake\View\View->render
CORE/src/Controller/Controller.php, line 617
⟩ Cake\Controller\Controller->render
CORE/src/Http/ActionDispatcher.php, line 128
⟩ Cake\Http\ActionDispatcher->_invoke
CORE/src/Http/ActionDispatcher.php, line 96
⟩ Cake\Http\ActionDispatcher->dispatch
CORE/src/Http/BaseApplication.php, line 83
⟩ Cake\Http\BaseApplication->__invoke
CORE/src/Http/Runner.php, line 65
⟩ Cake\Http\Runner->__invoke
CORE/src/Routing/Middleware/RoutingMiddleware.php, line 62
⟩ Cake\Routing\Middleware\RoutingMiddleware->__invoke
CORE/src/Http/Runner.php, line 65
⟩ Cake\Http\Runner->__invoke
CORE/src/Routing/Middleware/AssetMiddleware.php, line 88
⟩ Cake\Routing\Middleware\AssetMiddleware->__invoke
CORE/src/Http/Runner.php, line 65
⟩ Cake\Http\Runner->__invoke
ROOT/vendor/markstory/asset_compress/src/Middleware/AssetCompressMiddleware.php, line 56
⟩ AssetCompress\Middleware\AssetCompressMiddleware->__invoke
CORE/src/Http/Runner.php, line 65
⟩ Cake\Http\Runner->__invoke
CORE/src/Error/Middleware/ErrorHandlerMiddleware.php, line 81
⟩ Cake\Error\Middleware\ErrorHandlerMiddleware->__invoke
CORE/src/Http/Runner.php, line 65
⟩ Cake\Http\Runner->__invoke
CORE/src/Http/Runner.php, line 51
⟩ Cake\Http\Runner->run
CORE/src/Http/Server.php, line 92
⟩ Cake\Http\Server->run
ROOT/webroot/index.php, line 37
```

**Comment by markstory** [comment]
Thanks @xrep I'll see if I can reproduce this issue.

**Comment by xrep** [comment]
Hi markstory, 

to reproduce this, I've installed **jquery** throug NPM, so npm creates in my `/app/node_modules/jquery `required files. Then I set asset_compress.ini file, and built through CLI build cached file  in `app/webroot/cached` 

Now, I've tried to create some custom javascript code in app/webroot/js/lol.js, then I've updated my asset_compress.ini with following code

```
[lulz.js]
files[] = lol.js
```

build was fine and with this I was able to call this file through

`<?= $this->AssetCompress->script('Admin.lulz.js'); ?>`

without exception.

**Comment by markstory** [comment]
What does the `paths[]` section of your config file look like? Given that you've installed jquery via npm you'd need to have `ROOT/node_modules/*` and not `node_modules/*` as that is not an absolute path.

**Comment by markstory** [comment]
Closing as I was able to reproduce this issue and it is a path resolution issue caused by 

```ini
[js]
timestamp = true
paths[] = node_modules/*
```

The `paths` config options need to be _absolute_ paths, which this path isn't. It works in the CLI tool because you are getting lucky as the current working directory is likely the application root. The same is not true for the webserver. By making that section look like

```
[js]
timestamp = true
paths[] = ROOT/node_modules/*
```

the issue went away for me.
